src/SFA.DAS.Employer.PR.Web/Controllers/Requests/ChangeNameController.cs

```diff
@@ -0,0 +1,83 @@
+﻿using FluentValidation;
+using FluentValidation.AspNetCore;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using SFA.DAS.Employer.PR.Domain.Interfaces;
+using SFA.DAS.Employer.PR.Domain.OuterApi.Responses;
+using SFA.DAS.Employer.PR.Web.Extensions;
+using SFA.DAS.Employer.PR.Web.Infrastructure;
+using SFA.DAS.Employer.PR.Web.Infrastructure.Services;
+using SFA.DAS.Employer.PR.Web.Models.Requests;
+using SFA.DAS.Employer.PR.Web.Models.Session;
+
+namespace SFA.DAS.Employer.PR.Web.Controllers.Requests;
+
+[Route("requests")]
+public class ChangeNameController(IOuterApiClient _outerApiClient, ISessionService _sessionService, IValidator<EmployerAccountNamesSubmitModel> _validator) : Controller
+{
+    public const string UserEmailDoesNotMatchRequestShutterPageViewPath = "~/Views/Requests/UserEmailDoesNotMatchRequestShutterPage.cshtml";
+    public const string RequestsChangeNameViewPath = "~/Views/Requests/CreateServiceAccountChangeName.cshtml";
+
+    [Authorize]
+    [HttpGet]
+    [Route("{requestId:guid}/createaccount/changename", Name = RouteNames.CreateAccountChangeName)]
+    public async Task<IActionResult> Index(Guid requestId, CancellationToken cancellationToken)
+    {
+        GetPermissionRequestResponse permissionRequest = await _outerApiClient.GetPermissionRequest(requestId, cancellationToken);
+
+        if (User.GetEmail() != permissionRequest.EmployerContactEmail) return View(UserEmailDoesNotMatchRequestShutterPageViewPath);
+
+        var sessionModel = _sessionService.Get<AccountCreationSessionModel>();
+        SetNamesInSessionModel(sessionModel, permissionRequest);
```

Comment:
What if session model is not there?

Reply:
The method doesn't try to update firstname/lastname from the session model.